language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ForeachUtil.java

```diff
@@ -0,0 +1,189 @@
+        String type = getTypeOfIteratorVariable(symbol);
+        StringBuilder snippet = new StringBuilder("foreach");
+        snippet.append(" ").append(type).append(" ")
+                .append(CommonUtil.getValidatedSymbolName(ctx, VAR_NAME)).append(" in ").append(symbolName)
+                .append(" ").append("{").append(CommonUtil.LINE_SEPARATOR).append("\t${1:}")
+                .append(CommonUtil.LINE_SEPARATOR).append("}");
+
+        //create the text edit item to replace the field accession text
+        List<TextEdit> textEdits = new ArrayList<>();
+        TextEdit textEdit = new TextEdit();
+        textEdit.setNewText("");
+        Position start = new Position(expr.lineRange().startLine().line(), expr.lineRange().startLine().offset());
+        Position end = new Position(expr.lineRange().endLine().line(), expr.lineRange().endLine().offset());
+        textEdit.setRange(new Range(start, end));
+        textEdits.add(textEdit);
+        CompletionItem completionItem =
+                ForeachCompletionItemBuilder.build(snippet.toString(), label, detail, textEdits);
+        return new StaticCompletionItem(ctx, completionItem, StaticCompletionItem.Kind.OTHER);
+    }
+
+    /**
+     * Returns the type of the member type of an iterable type symbol.
+     *
+     * @param symbol iterable type symbol.
+     * @return signature of the member type.
+     */
+    public static String getTypeOfIteratorVariable(TypeSymbol symbol) {
+        String type;
+        switch (symbol.typeKind()) {
+            case STRING:
+                type = symbol.typeKind().getName();
+                break;
+            case ARRAY:
+                type = ((ArrayTypeSymbol) symbol).memberTypeDescriptor().signature();
+                break;
+            case TUPLE:
+                List<String> typesSet = new ArrayList<>(((TupleTypeSymbol) symbol).
+                        memberTypeDescriptors().stream().map(
+                        tSymbol -> tSymbol.typeKind().getName()).collect(Collectors.toSet()));
+                if (typesSet.size() == 1) {
+                    type = typesSet.get(0);
+                } else {
+                    type = VAR_TYPE;
+                }
+                break;
+            case MAP:
+                type = ((MapTypeSymbol) symbol).typeParam().signature();
+                break;
+            case RECORD:
+                //todo: Fix when #30245 is fixed.
+                type = VAR_TYPE;
+                break;
+            case TABLE:
+                type = ((TableTypeSymbol) symbol).rowTypeParameter().signature();
+                break;
+            case STREAM:
+                type = ((StreamTypeSymbol) symbol).typeParameter().signature();
+                break;
+            case XML:
+                type = symbol.signature();
+                break;
+            case TYPE_REFERENCE:
+                type = getTypeOfIteratorVariable(((TypeReferenceTypeSymbol) symbol).typeDescriptor());
```

Comment:
You can avoid this by using the `CommonUtil.getRawType` API. Instead of doing the type check for the type symbol in this switch-case, let's do the check against the raw type.

Reply:
Done!